src/title.rs

```diff
@@ -0,0 +1,91 @@
+use crate::{
+    layout::Alignment,
+    text::{Line, Span},
+};
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Title<'a> {
+    pub content: Option<Line<'a>>,
+    pub alignment: Alignment,
+    pub position: Position,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum Position {
+    Top,
+    Bottom,
+}
+
+impl<'a> Title<'a> {
+    pub fn content<T>(mut self, content: T) -> Title<'a>
+    where
+        T: Into<Line<'a>>,
+    {
+        self.content = Some(content.into());
+        self
+    }
+
+    pub fn alignment(mut self, alignment: Alignment) -> Title<'a> {
+        self.alignment = alignment;
+        self
+    }
+
+    pub fn position(mut self, position: Position) -> Title<'a> {
+        self.position = position;
+        self
+    }
+}
+
+impl<'a> From<Line<'a>> for Title<'a> {
+    fn from(value: Line<'a>) -> Self {
+        Self::default().content(value)
+    }
+}
+
+impl<'a> From<String> for Title<'a> {
+    fn from(value: String) -> Self {
+        Self::default().content(value)
+    }
+}
+
+impl<'a> From<&'a str> for Title<'a> {
+    fn from(value: &'a str) -> Self {
+        Self::default().content(value)
+    }
+}
+
+impl<'a> From<Vec<Span<'a>>> for Title<'a> {
+    fn from(value: Vec<Span<'a>>) -> Self {
+        Self::default().content(value)
+    }
+}
+
+impl<'a> From<Span<'a>> for Title<'a> {
+    fn from(span: Span<'a>) -> Self {
+        Self::from(vec![span])
+    }
+}
+
+impl<'a> From<Title<'a>> for String {
```

Comment:
https://github.com/tui-rs-revival/ratatui/blob/593fd29d00dde265be3dd83cc8c7ca132022cf87/src/text/line.rs#L129-L136
Not sure, I saw it implemented for Line so I thought it might be needed here as well. I removed it for now.